app/js/lib/speech/synthesis.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const p = Object.freeze({
+  // Properties
+  synthesis: Symbol('synthesis'),
+  supportsSynthesis: Symbol('supportsSynthesis'),
+  voice: Symbol('voice'),
+
+  // Methods
+  populateVoice: Symbol('populateVoice'),
+});
+
+export default class SpeechSynthesis {
+  constructor() {
+    const synthesis = window.speechSynthesis;
+
+    this[p.supportsSynthesis] = !!synthesis;
+
+    if (this[p.supportsSynthesis]) {
+      this[p.synthesis] = synthesis;
+      this[p.synthesis].onvoiceschanged = this[p.populateVoice].bind(this);
+    } else {
+      this[p.synthesis] = null;
+    }
+
+    this[p.voice] = null;
+
+    Object.seal(this);
+  }
+
+  /**
+   * Speak a text aloud.
+   *
+   * @param {string} text
+   */
+  speak(text = '') {
+    if (!text) {
```

Comment:
The value passed to `SpeechSynthesisUtterance` is converted to string, so it's already resilient enough not to throw an error. The test here is to shortcut the function if the text is empty.
But I agree that it could be symptomatic of a bug in the application.